src/app.js

```diff
@@ -1,13 +1,258 @@
+  $('#trip-headers').html('');
+  clearContent();
+  tableFields.forEach((field) => $('#trip-headers').append(tripHeadersTemplate({header: field})));
+  renderTrips(list);
+  loadFilters();
+};
+
+const loadFilters = function loadFilters() {
+  $('#filter-select').empty();
+  $('#filter').show();
+  tableFields.forEach((filter) => {
+    const capitalFilter = filter.charAt(0).toUpperCase() + filter.slice(1);
+    $('#filter-select').append(filterTemplate({item: filter, capitalized: capitalFilter}))
+  });
+  $('#filter-select').prop('selectedIndex', -1);
+};
+
+const setFilterCategory = function setFilterCategory() {
+  const capital = $('#filter-select').find(':selected').text();
+  filterCategory = capital.charAt(0).toLowerCase() + capital.slice(1);
+};
+
+const filterTrips = function filterTrips(event) {
+  let filterInput = event.currentTarget.value;
+  let filteredTrips = [];
+  switch (filterCategory) {
+    case 'name':
+    case 'category':
+    case 'continent':
+      filterInput = filterInput.toLowerCase();
+      filteredTrips = tripList.models.filter(trip => trip.attributes[filterCategory].toLowerCase().includes(filterInput));
+      referenceList.reset(filteredTrips);
+      renderTrips(referenceList);
+      break;
+    case 'weeks':
+    case 'cost':
+      filterInput = parseFloat(filterInput);
+      filteredTrips = tripList.models.filter(trip => trip.attributes[filterCategory] <= filterInput);
+      referenceList.reset(filteredTrips);
+      renderTrips(referenceList);
+      break;
+    default:
+      break;
+  }
+};
+
+const splitScreen = function splitScreen() {
+  $('.left-half').addClass('columns small-6 float-left');
+  $('.right-half').addClass('columns small-6 float-right');
+};
+
+const unSplitScreen = function unSplitScreen() {
+  $('.left-half, .right-half').removeClass('columns small-6 float-left');
+  $('#filter-input').val('');
+}
+
+const highlightCurrentTrip = function highlightCurrentTrip(event) {
+  const currentRow = $(event.currentTarget).parent().parents('tr:first').children();
+  $('#all-trips tr td').removeClass('current-trip');
+  $(currentRow).addClass('current-trip');
+};
+
+const showTrip = function showTrip(event) {
+  $('#show-trip').addClass('show-right').siblings().removeClass('show-right');
+  highlightCurrentTrip(event);
+  clearContent();
+  splitScreen();
+  const tripUrl = `${url}/${event.currentTarget.id}`;
+  $.get(tripUrl, (response) => {
+    response.plural = (response.weeks === 1) ? '' : 's';
+    $('#show-trip').append(showTripTemplate(response));
+  })
+};
+
+const clearContent = function clearContent() {
+  $('.content').empty();
+};
+
+const clearMessages = function clearMessages() {
+  $('#status-messages ul').empty();
+  $('#status-messages').hide();
+};
+
+const loadFormFields = function loadFormFields(formId, formFieldsArray) {
+  clearContent();
+  formFieldsArray.forEach((item) => {
+    $(`#${formId}`).append(formTemplate({field: item}));
```

Comment:
This is a very clever method to generate the form you need.  The downside is that you can't create controls like `select` which keep a user from entering an invalid choice.  Similarly you can't have inputs of type `number`.  

How could you adapt this method to improve the UX?